.github/workflows/oaf-check.yml

```diff
@@ -14,8 +20,10 @@ jobs:
       - uses: actions/setup-python@v5
         with:
           python-version: '3.11'
+          cache: 'pip'
+          cache-dependency-path: 'requirements/dev.txt'
       - name: Install dependencies
-        run: pip install -U pip-tools
+        run: pip install -r requirements/dev.txt # use the same pip-tools as dev
```

Comment:
We could do something like this to avoid installing unnecessary dependencies, `pip` isn't the fastest

```
pip install $(grep "pip-tools==" requirements/dev.txt)
```